src/Timer/Subscription.workflow.cs

```diff
@@ -0,0 +1,22 @@
+namespace TemporalioSamples.Timer;
+
+using Microsoft.Extensions.Logging;
+using Temporalio.Workflows;
+
+[Workflow]
+public class Subscription
+{
+    [WorkflowRun]
+    public async Task RunAsync(string userId)
+    {
+        while (true)
+        {
+            await Workflow.DelayAsync(TimeSpan.FromDays(30));
```

Comment:
Arguably it's not worth a whole example to demonstrate using `DelayAsync` (I am not sure any other SDK has such a simple timer sample)

Reply:
I think it will be helpful to some, particularly those new to temporal. I'd point someone to this to show what I meant by "sleep for days" while pitching.

I also added a bit on handling workflow cancellation

Reply:
There is a case where too many samples of too trivial things can be present. Not sure other SDK samples repositories would have accepted such trivial timer sample, unsure if .NET should be the inconsistent leader here.

Reply:
All but .net/python have at least one mention of timer in the readme's list of samples, and I didn't find a use of delayasync in this repo. I think basic samples are valuable, and imagine they're easy to skip over for those who are more experienced and know what a timer is.

Reply:
:+1: Those other samples do seem more involved (maybe w/ the exception of Go). Would you be willing to make this sample at least as robust as the TypeScript one?

Reply:
Would like to, but don't have time to, sorry!